types/assert-equal-jsx/assert-equal-jsx-tests.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import {assertEqualJSX} from 'assert-equal-jsx';
```

Comment:
TypeScript won't give you a compile error if the import matches the declared export, but the code still won't work at runtime if the declared export is wrong. So intellisense working doesn't mean that the definition is correct. You have to actually run your code to see that it works.
Note that you don't have to put something on DefinitelyTyped to start using a new library. It's better to [test](https://github.com/DefinitelyTyped/DefinitelyTyped#test-a-new-package) out your definitions on your own and contribute them after you've successfully used them in a working program.